### PR TITLE
[api] fix gas estimation calc bug

### LINKED ISSUE #2761: Gas Estimation failed when calling selfdestruct()
### What did you do? If possible, provide a recipe for reproducing the error.

To resolve #2618, I deployed a simple contract on the chain for testing.

```solidity
pragma solidity >0.6.0;
contract Storage {
 address payable private owner;
 uint256 number;
 constructor() {
  owner = payable(msg.sender);
 }
 function store(uint256 num) public {
  number = num;
 }
 function retrieve() public view returns (uint256){
  return number;
 }
 
 function close() public { 
  selfdestruct(owner); 
 }
}

```

However, it failed to calling close() function which executes selfdestruct action of the contract.

### What did you expect to see?

smart contract executes selfdestruct() action, and it transfers its token to the address of the owner

### What did you see instead?

error message:

> Internal JSON-RPC error. { "code": 13, "message": "Out of gas" }

### PR DESCRIPTION
**Pull request by Liuhaai**
fixed #2761 

## Causes: selfdestruct invokes gas refund policy
In most of time, refund is 0 because stateDB.GetRefund() returns 0. However, when selfdestruct behavior happens, refund value will not be `stateDB.GetRefund()`, but `(evmParams.gas - remainingGas) / 2`. This is because 
> Ethereum refunds up to half of the gas used by a contract transaction calling the selfdestruct function to the transaction sender.

```
refund := (evmParams.gas - remainingGas) / 2
if refund > stateDB.GetRefund() {
	refund = stateDB.GetRefund()
}
remainingGas += refund
```

When `estimatedGas` is evaluated second time, `OutofGas` err will be returned because `remainingGas < intriGas`.

### CODE REVIEW

**Comment by dustinxie** [comment]
we discussed this issue earlier, 21000 is a ETH-specific setting, and we do the gas cap in babel -- if estimate < 21000, babel will return 21000.

so we don't need this in iotex-core?

File: action/protocol/execution/evm/evm.go [hunk 1]
```diff
@@ -291,7 +291,7 @@ func executeInEVM(evmParams *Params, stateDB *StateDBAdapter, g genesis.Blockcha
 		return nil, evmParams.gas, remainingGas, action.EmptyAddress, uint64(iotextypes.ReceiptStatus_Failure), err
 	}
 	if remainingGas < intriGas {
-		return nil, evmParams.gas, remainingGas, action.EmptyAddress, uint64(iotextypes.ReceiptStatus_Failure), action.ErrOutOfGas
+		return nil, evmParams.gas, remainingGas, action.EmptyAddress, uint64(iotextypes.ReceiptStatus_ErrOutOfGas), nil
```
Comment (dustinxie) [inline action/protocol/execution/evm/evm.go:294]: this would cause a fork?

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.14-alpine as build
+FROM golang:1.17 as build
```
Comment (dustinxie) [inline Dockerfile:1]: should go to PR2784

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/iotexproject/iotex-core
 
-go 1.13
+go 1.17
```
Comment (dustinxie) [inline go.mod:3]: should go to PR2784

File: go.mod [hunk 4]
```diff
@@ -47,6 +47,126 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 )
 
+require (
```
Comment (dustinxie) [inline go.mod:50]: should go to PR2784

File: api/api_test.go [hunk 5]
```diff
@@ -2762,3 +2762,37 @@ func TestServer_GetActPoolActions(t *testing.T) {
...
+	require := require.New(t)
+	cfg := newConfig(t)
+
+	svr, bfIndexFile, err := createServer(cfg, true)
+	require.NoError(err)
+	defer func() {
+		testutil.CleanupPath(t, bfIndexFile)
+	}()
+
+	// deploy self-desturct contract
```
Comment (dustinxie) [inline api/api_test.go:2776]: i may miss the context, what's special about this test?

couple of days ago, there's already a check-in about self-destruct https://github.com/iotexproject/iotex-core/commit/dcf27b4902cebda891b1edbf5d817515f933bb55

Comment (dustinxie) [reply api/api_test.go:2776]: discussed with Haixiang, fix should go into `api.go`

**Comment by dustinxie** [comment]
can you revert the go 1.17 changes?
that should be another PR

**Comment by Liuhaai** [comment]
> can you revert the go 1.17 changes?
> that should be another PR

Good

**Review by dustinxie — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

**Pull request merged**